src/shell/output.c

```diff
@@ -78,6 +81,65 @@ static void shell_output_control_task (struct shell_task *task,
     }
 }
 
+static void shell_output_ready_completion (flux_future_t *f, void *arg)
+{
+    struct shell_output *out = arg;
+
+    out->output_ready = true;
+    flux_future_destroy (out->output_ready_f);
+    out->output_ready_f = NULL;
+}
+
+/* log entry to exec.eventlog that we've created the output directory */
+static int shell_output_ready (struct shell_output *out)
+{
+    int saved_errno;
+    flux_kvs_txn_t *txn = NULL;
+    json_t *entry = NULL;
+    char *entrystr = NULL;
+    const char *key = "exec.eventlog";
+    int rc = -1;
+
+    if (!(entry = eventlog_entry_create (0., "output-ready", NULL))) {
+        log_err ("eventlog_entry_create");
+        goto error;
+    }
+    if (!(entrystr = eventlog_entry_encode (entry))) {
+        log_err ("eventlog_entry_encode");
+        goto error;
+    }
+    if (!(txn = flux_kvs_txn_create ())) {
+        log_err ("flux_kvs_txn_create");
+        goto error;
+    }
+    if (flux_kvs_txn_put (txn, FLUX_KVS_APPEND, key, entrystr) < 0) {
+        log_err ("flux_kvs_txn_put");
+        goto error;
+    }
+    if (!(out->output_ready_f = flux_kvs_commit (out->shell->h,
```

Comment:
This should go in the same commit as the output header (and probably the first data entry).